File: 02_algorithm/programmers/Level3/멀리 뛰기.py
def solution(n):
    DP = [1, 2]
    i = 0
    while len(DP) < n:
        DP.append(DP[i] + DP[i + 1])
        i += 1
    return DP[n - 1] % 1234567

print(solution(4))

File: 06_js/04_algorithm/programmers/Level3/멀리 뛰기.js
function solution(n) {
  let arr = [1, 2];
  let i = 0;
  while (arr.length < n) {
    arr.push(arr[i] % 1234567 + arr[i + 1] % 1234567);
    i += 1;
  }
  return arr[n - 1] % 1234567;
}

console.log(solution(4)); // 5
console.log(solution(3)); // 3